pragma-oracle/src/oracle/oracle.cairo

```diff
@@ -1771,6 +1824,37 @@ mod Oracle {
             }
         }
 
+        // @notice register a new tokenized vault into the regisry (priced with STRK)
+        // @dev Callable only by the owner
+        // @dev the token must be registered as currency and pair in the oracle registry
+        // @dev We reserve the owner of the contract the right to overwrite an existing token address
+        // @param token The token to register
+        // @param token_address Token address to register
+        fn register_tokenized_vault(
+            ref self: ContractState, token: felt252, token_address: ContractAddress
+        ) {
+            OracleInternal::assert_only_admin();
+            assert(token != 0, 'Token cannot be 0');
+            assert(
+                token_address != starknet::contract_address_const::<0>(),
+                'Token address cannot be 0'
+            );
+            self.tokenized_vault.write((token, 'STRK'), token_address)
+        }
+
+        // @notice delete a registered tokenized vault 
+        // @dev Callable only by the admin
+        // @param token The token to be removed
+        fn delete_tokenized_vault(ref self: ContractState, token: felt252) {
```

Comment:
can't we just use the register_tokenized_vault and allow null address ?

Reply:
yes it's possible, it might be even simpler